### PR TITLE
Add API packages

### PR DESCRIPTION
**Pull request by JRoy**

### CODE REVIEW

File: Essentials/src/main/java/net/essentialsx/api/v2/package-info.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/**
+ * API Classes, Services, and Events provided by EssentialsX v2.x
```
Comment (mdcfe) [inline Essentials/src/main/java/net/essentialsx/api/v2/package-info.java]:
```suggestion
 * API classes, services, and events provided by EssentialsX 2.x.
```

File: Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package net.essentialsx.api.v2.services;
+
+/**
+ * Provides basic information about a service registered by EssentialsX.
```
Comment (mdcfe) [inline Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java]:
```suggestion
 * An EssentialsX service that provides basic information about itself.
```

File: Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ */
+public interface EssentialsService {
+    /**
+     * Gets the name of this service.
+     * @return The name of the service.
+     */
+    String getName();
```
Comment (mdcfe) [inline Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java]: What exactly is this supposed to convey or be used for? The name of the service interface should identify the service, and ideally the implementation shouldn't be exposed to API users, so what does this represent?

File: Essentials/src/main/java/net/essentialsx/api/v2/services/package-info.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/**
+ * APIs that are registered with Bukkit's ServicesManager and can be accessed without touching Essentials internals.
```
Comment (mdcfe) [inline Essentials/src/main/java/net/essentialsx/api/v2/services/package-info.java]:
```suggestion
 * API services provided by EssentialsX. These are registered with Bukkit's ServicesManager, and can be accessed from there without getting an instance of EssentialsX first.
```
Should probably `{@link ServicesManager}` too

**Review by mdcfe — CHANGES_REQUESTED** [review]

File: Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java (hunk 3 above)
Comment (JRoy) [reply Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java]: I don't remember why I added this, will remove

File: Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package net.essentialsx.api.v2.services;
+
+/**
+ * An EssentialsX service that provides basic information about itself.
```
Comment (mdcfe) [inline Essentials/src/main/java/net/essentialsx/api/v2/services/EssentialsService.java]:
```suggestion
 * A service provided by EssentialsX.
```
Not sure if we actually need to keep this interface at all? (Yes I probably should've brought that up before, don't judge me)

**Review by mdcfe — APPROVED** [review]

**Pull request merged**